File: binding/include/network/libqrestaccessmanager.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qrestaccessmanager.html
pub const qrestaccessmanager = struct {

    /// New constructs a new QRestAccessManager object.
    pub fn New(manager: ?*anyopaque) QtC.QRestAccessManager {
        return qtc.QRestAccessManager_new(@ptrCast(manager));
    }


    /// New2 constructs a new QRestAccessManager object.
    pub fn New2(manager: ?*anyopaque, parent: ?*anyopaque) QtC.QRestAccessManager {
        return qtc.QRestAccessManager_new2(@ptrCast(manager), @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QRestAccessManager_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qrestaccessmanager.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn NetworkAccessManager(self: ?*anyopaque, ) QtC.QNetworkAccessManager {
        return qtc.QRestAccessManager_NetworkAccessManager(@ptrCast(self));
    }

    pub fn DeleteResource(self: ?*anyopaque, request: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_DeleteResource(@ptrCast(self), @ptrCast(request));
    }

    pub fn OnDeleteResource(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_DeleteResource(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Head(self: ?*anyopaque, request: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Head(@ptrCast(self), @ptrCast(request));
    }

    pub fn OnHead(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Head(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Get(self: ?*anyopaque, request: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Get(@ptrCast(self), @ptrCast(request));
    }

    pub fn OnGet(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Get(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Get2(self: ?*anyopaque, request: ?*anyopaque, data: []u8) QtC.QNetworkReply {
        const data_str = qtc.libqt_string{
    .len = data.len,
    .data = data.ptr,
};
return qtc.QRestAccessManager_Get2(@ptrCast(self), @ptrCast(request), data_str);
    }

    pub fn OnGet2(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, [*:0]u8) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Get2(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Get3(self: ?*anyopaque, request: ?*anyopaque, data: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Get3(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnGet3(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Get3(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Get4(self: ?*anyopaque, request: ?*anyopaque, data: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Get4(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnGet4(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Get4(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Post(self: ?*anyopaque, request: ?*anyopaque, data: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Post(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnPost(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Post(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Post2(self: ?*anyopaque, request: ?*anyopaque, data: *const QMap<QString, QVariant>) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Post2(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnPost2(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, *const QMap<QString, QVariant>) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Post2(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Post3(self: ?*anyopaque, request: ?*anyopaque, data: []u8) QtC.QNetworkReply {
        const data_str = qtc.libqt_string{
    .len = data.len,
    .data = data.ptr,
};
return qtc.QRestAccessManager_Post3(@ptrCast(self), @ptrCast(request), data_str);
    }

    pub fn OnPost3(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, [*:0]u8) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Post3(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Post4(self: ?*anyopaque, request: ?*anyopaque, data: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Post4(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnPost4(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Post4(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Post5(self: ?*anyopaque, request: ?*anyopaque, data: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Post5(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnPost5(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Post5(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Put(self: ?*anyopaque, request: ?*anyopaque, data: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Put(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnPut(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Put(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Put2(self: ?*anyopaque, request: ?*anyopaque, data: *const QMap<QString, QVariant>) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Put2(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnPut2(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, *const QMap<QString, QVariant>) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Put2(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Put3(self: ?*anyopaque, request: ?*anyopaque, data: []u8) QtC.QNetworkReply {
        const data_str = qtc.libqt_string{
    .len = data.len,
    .data = data.ptr,
};
return qtc.QRestAccessManager_Put3(@ptrCast(self), @ptrCast(request), data_str);
    }

    pub fn OnPut3(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, [*:0]u8) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Put3(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Put4(self: ?*anyopaque, request: ?*anyopaque, data: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Put4(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnPut4(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Put4(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Put5(self: ?*anyopaque, request: ?*anyopaque, data: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Put5(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnPut5(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Put5(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Patch(self: ?*anyopaque, request: ?*anyopaque, data: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Patch(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnPatch(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Patch(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Patch2(self: ?*anyopaque, request: ?*anyopaque, data: *const QMap<QString, QVariant>) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Patch2(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnPatch2(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, *const QMap<QString, QVariant>) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Patch2(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Patch3(self: ?*anyopaque, request: ?*anyopaque, data: []u8) QtC.QNetworkReply {
        const data_str = qtc.libqt_string{
    .len = data.len,
    .data = data.ptr,
};
return qtc.QRestAccessManager_Patch3(@ptrCast(self), @ptrCast(request), data_str);
    }

    pub fn OnPatch3(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, [*:0]u8) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Patch3(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Patch4(self: ?*anyopaque, request: ?*anyopaque, data: ?*anyopaque) QtC.QNetworkReply {
        return qtc.QRestAccessManager_Patch4(@ptrCast(self), @ptrCast(request), @ptrCast(data));
    }

    pub fn OnPatch4(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_Patch4(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn SendCustomRequest(self: ?*anyopaque, request: ?*anyopaque, method: []u8, data: []u8) QtC.QNetworkReply {
        const method_str = qtc.libqt_string{
    .len = method.len,
    .data = method.ptr,
};
const data_str = qtc.libqt_string{
    .len = data.len,
    .data = data.ptr,
};
return qtc.QRestAccessManager_SendCustomRequest(@ptrCast(self), @ptrCast(request), method_str, data_str);
    }

    pub fn OnSendCustomRequest(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, [*:0]u8, [*:0]u8) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_SendCustomRequest(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn SendCustomRequest2(self: ?*anyopaque, request: ?*anyopaque, method: []u8, data: ?*anyopaque) QtC.QNetworkReply {
        const method_str = qtc.libqt_string{
    .len = method.len,
    .data = method.ptr,
};
return qtc.QRestAccessManager_SendCustomRequest2(@ptrCast(self), @ptrCast(request), method_str, @ptrCast(data));
    }

    pub fn OnSendCustomRequest2(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, [*:0]u8, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_SendCustomRequest2(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn SendCustomRequest3(self: ?*anyopaque, request: ?*anyopaque, method: []u8, data: ?*anyopaque) QtC.QNetworkReply {
        const method_str = qtc.libqt_string{
    .len = method.len,
    .data = method.ptr,
};
return qtc.QRestAccessManager_SendCustomRequest3(@ptrCast(self), @ptrCast(request), method_str, @ptrCast(data));
    }

    pub fn OnSendCustomRequest3(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, [*:0]u8, ?*anyopaque) callconv(.c) void) void {
        qtc.QRestAccessManager_Connect_SendCustomRequest3(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QRestAccessManager_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qrestaccessmanager.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QRestAccessManager_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qrestaccessmanager.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QRestAccessManager_Delete(@ptrCast(self));
    }
};

File: binding/src/network/libqrestaccessmanager.cpp
#include <QByteArray>
#include <QDebug>
#include <QHttpMultiPart>
#include <QIODevice>
#include <QJsonDocument>
#include <QNetworkAccessManager>
#include <QNetworkReply>
#include <QNetworkRequest>
#include <QObject>
#include <QRestAccessManager>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qrestaccessmanager.h>
#include "libqrestaccessmanager.h"
#include "libqrestaccessmanager.hxx"

QRestAccessManager* QRestAccessManager_new(QNetworkAccessManager* manager) {
	 return new QRestAccessManager(manager);
}

QRestAccessManager* QRestAccessManager_new2(QNetworkAccessManager* manager, QObject* parent) {
	 return new QRestAccessManager(manager, parent);
}

libqt_string QRestAccessManager_Tr(const char* s) {
	QString _ret = QRestAccessManager::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QNetworkAccessManager* QRestAccessManager_NetworkAccessManager(const QRestAccessManager* self) {
	return self->networkAccessManager();
}

QNetworkReply* QRestAccessManager_DeleteResource(QRestAccessManager* self, const QNetworkRequest* request) {
	return self->deleteResource(*request);
}

void QRestAccessManager_Connect_DeleteResource(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::deleteResource, [self, slotFunc](const QNetworkRequest& request) {
	slotFunc(self, request);
    });
}

QNetworkReply* QRestAccessManager_Head(QRestAccessManager* self, const QNetworkRequest* request) {
	return self->head(*request);
}

void QRestAccessManager_Connect_Head(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::head, [self, slotFunc](const QNetworkRequest& request) {
	slotFunc(self, request);
    });
}

QNetworkReply* QRestAccessManager_Get(QRestAccessManager* self, const QNetworkRequest* request) {
	return self->get(*request);
}

void QRestAccessManager_Connect_Get(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::get, [self, slotFunc](const QNetworkRequest& request) {
	slotFunc(self, request);
    });
}

QNetworkReply* QRestAccessManager_Get2(QRestAccessManager* self, const QNetworkRequest* request, const libqt_string data) {
	return self->get(*request, QByteArray(data.data, data.len));
}

void QRestAccessManager_Connect_Get2(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, const char*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, const char*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::get, [self, slotFunc](const QNetworkRequest& request, const QByteArray& data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Get3(QRestAccessManager* self, const QNetworkRequest* request, const QJsonDocument* data) {
	return self->get(*request, *data);
}

void QRestAccessManager_Connect_Get3(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, const QJsonDocument*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, const QJsonDocument*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::get, [self, slotFunc](const QNetworkRequest& request, const QJsonDocument& data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Get4(QRestAccessManager* self, const QNetworkRequest* request, QIODevice* data) {
	return self->get(*request, data);
}

void QRestAccessManager_Connect_Get4(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, QIODevice*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, QIODevice*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::get, [self, slotFunc](const QNetworkRequest& request, QIODevice* data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Post(QRestAccessManager* self, const QNetworkRequest* request, const QJsonDocument* data) {
	return self->post(*request, *data);
}

void QRestAccessManager_Connect_Post(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, const QJsonDocument*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, const QJsonDocument*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::post, [self, slotFunc](const QNetworkRequest& request, const QJsonDocument& data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Post2(QRestAccessManager* self, const QNetworkRequest* request, const libqt_map data) {
	return self->post(*request, QMap<QString, QVariant>());
}

void QRestAccessManager_Connect_Post2(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, libqt_map) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, libqt_map)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::post, [self, slotFunc](const QNetworkRequest& request, const QMap<QString, QVariant>& data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Post3(QRestAccessManager* self, const QNetworkRequest* request, const libqt_string data) {
	return self->post(*request, QByteArray(data.data, data.len));
}

void QRestAccessManager_Connect_Post3(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, const char*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, const char*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::post, [self, slotFunc](const QNetworkRequest& request, const QByteArray& data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Post4(QRestAccessManager* self, const QNetworkRequest* request, QHttpMultiPart* data) {
	return self->post(*request, data);
}

void QRestAccessManager_Connect_Post4(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, QHttpMultiPart*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, QHttpMultiPart*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::post, [self, slotFunc](const QNetworkRequest& request, QHttpMultiPart* data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Post5(QRestAccessManager* self, const QNetworkRequest* request, QIODevice* data) {
	return self->post(*request, data);
}

void QRestAccessManager_Connect_Post5(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, QIODevice*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, QIODevice*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::post, [self, slotFunc](const QNetworkRequest& request, QIODevice* data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Put(QRestAccessManager* self, const QNetworkRequest* request, const QJsonDocument* data) {
	return self->put(*request, *data);
}

void QRestAccessManager_Connect_Put(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, const QJsonDocument*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, const QJsonDocument*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::put, [self, slotFunc](const QNetworkRequest& request, const QJsonDocument& data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Put2(QRestAccessManager* self, const QNetworkRequest* request, const libqt_map data) {
	return self->put(*request, QMap<QString, QVariant>());
}

void QRestAccessManager_Connect_Put2(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, libqt_map) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, libqt_map)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::put, [self, slotFunc](const QNetworkRequest& request, const QMap<QString, QVariant>& data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Put3(QRestAccessManager* self, const QNetworkRequest* request, const libqt_string data) {
	return self->put(*request, QByteArray(data.data, data.len));
}

void QRestAccessManager_Connect_Put3(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, const char*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, const char*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::put, [self, slotFunc](const QNetworkRequest& request, const QByteArray& data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Put4(QRestAccessManager* self, const QNetworkRequest* request, QHttpMultiPart* data) {
	return self->put(*request, data);
}

void QRestAccessManager_Connect_Put4(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, QHttpMultiPart*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, QHttpMultiPart*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::put, [self, slotFunc](const QNetworkRequest& request, QHttpMultiPart* data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Put5(QRestAccessManager* self, const QNetworkRequest* request, QIODevice* data) {
	return self->put(*request, data);
}

void QRestAccessManager_Connect_Put5(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, QIODevice*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, QIODevice*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::put, [self, slotFunc](const QNetworkRequest& request, QIODevice* data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Patch(QRestAccessManager* self, const QNetworkRequest* request, const QJsonDocument* data) {
	return self->patch(*request, *data);
}

void QRestAccessManager_Connect_Patch(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, const QJsonDocument*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, const QJsonDocument*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::patch, [self, slotFunc](const QNetworkRequest& request, const QJsonDocument& data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Patch2(QRestAccessManager* self, const QNetworkRequest* request, const libqt_map data) {
	return self->patch(*request, QMap<QString, QVariant>());
}

void QRestAccessManager_Connect_Patch2(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, libqt_map) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, libqt_map)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::patch, [self, slotFunc](const QNetworkRequest& request, const QMap<QString, QVariant>& data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Patch3(QRestAccessManager* self, const QNetworkRequest* request, const libqt_string data) {
	return self->patch(*request, QByteArray(data.data, data.len));
}

void QRestAccessManager_Connect_Patch3(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, const char*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, const char*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::patch, [self, slotFunc](const QNetworkRequest& request, const QByteArray& data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_Patch4(QRestAccessManager* self, const QNetworkRequest* request, QIODevice* data) {
	return self->patch(*request, data);
}

void QRestAccessManager_Connect_Patch4(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, QIODevice*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, QIODevice*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::patch, [self, slotFunc](const QNetworkRequest& request, QIODevice* data) {
	slotFunc(self, request, data);
    });
}

QNetworkReply* QRestAccessManager_SendCustomRequest(QRestAccessManager* self, const QNetworkRequest* request, const libqt_string method, const libqt_string data) {
	return self->sendCustomRequest(*request, QByteArray(method.data, method.len), QByteArray(data.data, data.len));
}

void QRestAccessManager_Connect_SendCustomRequest(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, const char*, const char*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, const char*, const char*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::sendCustomRequest, [self, slotFunc](const QNetworkRequest& request, const QByteArray& method, const QByteArray& data) {
	slotFunc(self, request, method, data);
    });
}

QNetworkReply* QRestAccessManager_SendCustomRequest2(QRestAccessManager* self, const QNetworkRequest* request, const libqt_string method, QIODevice* data) {
	return self->sendCustomRequest(*request, QByteArray(method.data, method.len), data);
}

void QRestAccessManager_Connect_SendCustomRequest2(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, const char*, QIODevice*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, const char*, QIODevice*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::sendCustomRequest, [self, slotFunc](const QNetworkRequest& request, const QByteArray& method, QIODevice* data) {
	slotFunc(self, request, method, data);
    });
}

QNetworkReply* QRestAccessManager_SendCustomRequest3(QRestAccessManager* self, const QNetworkRequest* request, const libqt_string method, QHttpMultiPart* data) {
	return self->sendCustomRequest(*request, QByteArray(method.data, method.len), data);
}

void QRestAccessManager_Connect_SendCustomRequest3(QRestAccessManager* self, intptr_t slot) {
    void (*slotFunc)(QRestAccessManager*, const QNetworkRequest*, const char*, QHttpMultiPart*) = reinterpret_cast<void (*)(QRestAccessManager*, const QNetworkRequest*, const char*, QHttpMultiPart*)>(slot);
    QRestAccessManager::connect(self, &QRestAccessManager::sendCustomRequest, [self, slotFunc](const QNetworkRequest& request, const QByteArray& method, QHttpMultiPart* data) {
	slotFunc(self, request, method, data);
    });
}

libqt_string QRestAccessManager_Tr2(const char* s, const char* c) {
	QString _ret = QRestAccessManager::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QRestAccessManager_Tr3(const char* s, const char* c, int n) {
	QString _ret = QRestAccessManager::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QRestAccessManager_Delete(QRestAccessManager* self) {
    delete self;
}

